### ISSUE TITLE
Binary XENT loss function incorrect

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/nd4j#1319]
https://en.wikipedia.org/wiki/Cross_entropy#Cross-entropy_error_function_and_logistic_regression

Should be: -1/n * sum( labels.muli(log(prob)).add((1-labels).muli(log(1-prob))

Old implementation looks a bit closer to what it should be:
https://github.com/deeplearning4j/nd4j/blob/4526c7dfcca0911cc79e217266eb4e7375e1eabb/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/lossfunctions/LossCalculation.java#L113

**Comment by eraly** [comment]
Merged. Closing.
